test/parse.jl

```diff
@@ -229,3 +229,21 @@ end
 @test tryparse(Float32, "1.23") === Nullable(1.23f0)
 @test tryparse(Float16, "1.23") === Nullable(Float16(1.23))
 
+# parsing complex numbers (#22250)
+@testset "complex parsing" begin
```

Comment:
It might also be good to add some tests for parsing bogus stuff, like `1+2ij` or `1im-3im`.

Reply:
Done.